march_hardware/include/march_hardware/MarchRobot.h

```diff
@@ -58,6 +61,16 @@ class MarchRobot
 
   Joint& getJoint(::std::string jointName);
 
+  Joint& getJoint(size_t index);
+
+  size_t size() const;
+
+  iterator begin();
+  iterator end();
+
+  const_iterator begin() const;
+  const_iterator end() const;
```

Comment:
The call doesn't have to be from a `const MarchRobot`. The `const` on the end means that the object is not modified during this function call and the `const_iterator` means the iterator iterates over `const Joint`, which do not allow you to modify them. For example, this is used when you want to loop over all joints to get their names (which is also constant). Furthermore, this allows the compiler to do some optimizations.

Reply:
But there exists a method begin() and a const method begin(). According to my limited understanding, the non-const method wil get called unless it is called from a const MarchRobot.

Reply:
I thought the `const_iterator` would be used for
```cpp
for (const auto& joint : march_robot)
{
...
}
```
But apparently it turns out [it doesn't](https://stackoverflow.com/questions/26554636/why-doesnt-const-range-based-for-use-const-iterator). However, what you are saying is not completely true, since you can use the const_iterator like so:
```cpp
march::MarchRobot robot(...);  // notice it is not const
march::MarchRobot::const_iterator it = robot.begin();
```
This is not the best use case since you should always use `for` loops. However, when you define this method you can also loop over joints when `MarchRobot` _is_ defined const. But, as you mentioned, we don't do that anywhere. So we could leave it in for the future, I think it could be useful for writing tests, where you create a robot and then only read, but it is definitely not necessary.

Reply:
In that case, I would remove it as it is not currently in use and has a small chance of being used (outside tests) in the future. At that point, one can always make these methods again.